src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -268,6 +271,52 @@ public bool TryParseEofHeader(ReadOnlySpan<byte> container, [NotNullWhen(true)]
 
         bool ValidateBody(ReadOnlySpan<byte> container, ref EofHeader? header)
         {
+            SectionHeader[]? codeSections = header.Value.CodeSections;
+            (int typeSectionStart, ushort typeSectionSize) = header.Value.TypeSection;
+            if (codeSections.Length == 0 || codeSections.Any(section => section.Size == 0))
+            {
+                if (_loggingEnabled)
+                {
+                    _logger.Trace($"EIP-3540 : CodeSection size must follow a CodeSection, CodeSection length was {codeSections.Length}");
+                }
+
+                header = null;
+                return false;
+            }
+
+            if (codeSections.Length != (typeSectionSize / MINIMUM_TYPESECTION_SIZE))
+            {
+                if (_loggingEnabled)
+                {
+                    _logger.Trace($"EIP-4750: Code Sections count must match TypeSection count, CodeSection count was {codeSections.Length}, expected {typeSectionSize / MINIMUM_TYPESECTION_SIZE}");
+                }
+
+                header = null;
+                return false;
+            }
+
+            if (container[typeSectionStart] != 0 && container[typeSectionStart] != 0)
```

Comment:
Looks like bug. I guess it should be
`container[typeSectionStart] != 0 && container[typeSectionStart+1] != 0`

Reply:
yes I was fixing it it should be ``container[typeSectionStart] != 0 || container[typeSectionStart] != 0``

Reply:
yes, || of course